src/App.tsx

```diff
@@ -15,48 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetical = 'Alphabetical',
+  Length = 'Length',
+}
+
+const sortGoods = (goodsArray: string[], sortType: SortType,
+  isReverse: boolean): string[] => {
+  let sortedGoods = [...goodsArray];
+
+  switch (sortType) {
+    case SortType.Alphabetical:
+      sortedGoods = sortedGoods.sort();
+      if (isReverse) {
+        sortedGoods.reverse();
+      }
+
+      return sortedGoods;
+    default:
+      sortedGoods = [...goodsFromServer].sort((a, b) => {
+        const lengthComparison = a.length - b.length;
+
+        if (lengthComparison === 0) {
+          return 1;
+        }
+
+        return lengthComparison;
+      });
+      if (isReverse) {
+        sortedGoods.reverse();
+      }
+
+      return sortedGoods;
+  }
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortType, setSortType] = useState<SortType | null>(null);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const handleSort = (type: SortType) => {
+    const sortedGoods = sortGoods([...goods], type, isReverse);
+
+    setGoods(sortedGoods);
+    setSortType(type);
+  };
+
+  const reverseGoods = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+    setIsReverse(!isReverse);
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setSortType(null);
+    setIsReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${(sortType === SortType.Alphabetical) ? '' : 'is-light'}`}
```

Comment:
```suggestion
          className={'button is-info', {"is-light": sortType === SortType.Alphabetical}}
```

Reply:
it's better to use classname library for making a dynamic classnames